riot.py: add distclean option and simplify code

haukepetersen opened the pull request:
Removed some code redundancy and added the option to distclean (useful when working with packages...).

miri64 commented on iotlab_controller/riot.py:
```diff
@@ -41,23 +41,19 @@ def path(self):
+                             self.application_path, "all"])
+
+    def clean(self, build_env=None, distclean=False):
+        op = "distclean" if distclean else "clean"
```
`distclean` does not imply `clean` (see https://github.com/RIOT-OS/RIOT/blob/55cc34fd174c241df69e9992024ed0590f2255f6/Makefile.include#L653-L655 vs. https://github.com/RIOT-OS/RIOT/blob/55cc34fd174c241df69e9992024ed0590f2255f6/Makefile.include#L640-L643), so this should rather be:

```suggestion
        op = ["distclean"] if distclean else ["distclean", "clean"]
```

(and the usage of op below adapted to `["make", "-C", self.application_path] + op`). However, I don't see why `distclean` should be within `clean`. Rather I would prefer a separate method.

haukepetersen replied:
ok, had it separately in my first iteration, as in my testing a `distclean` always removed my complste `app/bin` folder. But as you state, going back to that should be the saver way -> will fix.

miri64 replied:
I've seen the removal of `app/bin` in https://github.com/RIOT-OS/RIOT/blob/55cc34fd174c241df69e9992024ed0590f2255f6/Makefile.include#L653-L655, however I'm not quite sure why it is their... given RIOT's current directory structure (having packages checked out within `app/bin` the first line seems to be redundant then).

miri64 commented on iotlab_controller/riot.py:
```diff
@@ -41,23 +41,19 @@ def path(self):
             return self.flashfile
 
-    def build(self, build_env=None, threads=1):
+    def run(self, build_env, cmd):
```
Make this private please.

```suggestion
    def _run(self, build_env, cmd):
```

There is no meaning for this outside of the context of the public methods below.

haukepetersen replied:
ups, that was simply because of my newby python skills, did never look up how to declare functions private in python... Will of course fix.

miri64 replied:
It's more [by convention than real declaration](https://docs.python.org/3/tutorial/classes.html#private-variables), but linters will look for that so its good to keep this convention ;-).

haukepetersen commented:
pushed a makever:
- `_run()` is not private
- removed the distclean parameter from `clean()`
- added separate `distclean()` as static function. Rationale here is that the `distclean` target is effecting not only the board that is specified in a `RIOTFirmware` object, but all targets in a specific application dir. So I think it makes sense to call the `distclean` target statically on a selected path this way.

miri64 commented:
Squashed and merged into master.
